great_expectations/core/usage_statistics/anonymizers/base.py

```diff
@@ -0,0 +1,217 @@
+import logging
+from abc import ABC, abstractmethod, abstractstaticmethod
+from hashlib import md5
+from typing import Any, List, Optional, Tuple
+
+from great_expectations.core.usage_statistics.util import (
+    aggregate_all_core_expectation_types,
+)
+from great_expectations.util import load_class
+
+logger = logging.getLogger(__name__)
+
+
+class BaseAnonymizer(ABC):
+
+    # Any class that starts with this __module__ is considered a "core" object
+    CORE_GE_OBJECT_MODULE_PREFIX = "great_expectations"
+
+    CORE_GE_EXPECTATION_TYPES = aggregate_all_core_expectation_types()
+
+    def __init__(self, salt: Optional[str] = None) -> None:
+        if salt is not None and not isinstance(salt, str):
+            logger.error("invalid salt: must provide a string. Setting a random salt.")
+            salt = None
+        if salt is None:
+            import secrets
+
+            self._salt = secrets.token_hex(8)
+        else:
+            self._salt = salt
+
+    @abstractmethod
+    def anonymize(self, obj: Optional[object], **kwargs) -> Any:
+        raise NotImplementedError
+
+    @abstractstaticmethod
+    def can_handle(obj: Optional[object], **kwargs) -> bool:
+        raise NotImplementedError
```

Comment:
This is still the core part of this class - what does a class need to be able to do to be considered an anonymizer?

Note that all the other methods (outside a few key ones) have been refactored out into separate anonymizer classes like before.